phd-tests/framework/src/disk/mod.rs

```diff
@@ -212,11 +254,29 @@ impl DiskFactory {
             &binary_path.as_std_path(),
             &ports,
             &self.storage_dir,
-            Some(&artifact_path),
-            Some(guest_os),
+            artifact_path.as_ref(),
+            guest_os,
             self.log_mode,
         )
         .map(Arc::new)
         .map_err(Into::into)
     }
+
+    pub(crate) async fn create_in_memory_disk<'d>(
+        &self,
+        name: String,
+        source: &DiskSource<'d>,
+        readonly: bool,
+    ) -> Result<Arc<InMemoryDisk>, DiskError> {
+        let contents = match source {
+            DiskSource::Artifact(name) => {
+                let (path, _) = self.get_guest_artifact_info(name).await?;
+                std::fs::read(path)?
```

Comment:
nitpicky: perhaps worthwhile to include the path in the error here?

Reply:
Done in 85c0b95.